File: lab6/SensorExperimentApp/app/src/main/java/com/example/sensorexperimentapp/MainActivity.java
package com.example.sensorexperimentapp;

import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private SensorManager sensorManager;
    private TextView sensorListTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button detectSensorsButton = findViewById(R.id.detectSensorsButton);
        sensorListTextView =findViewById(R.id.sensorListTextView);

        sensorManager =(SensorManager) getSystemService(SENSOR_SERVICE);
        detectSensorsButton.setOnClickListener(v-> listAvailableSensors());
    }

    private void listAvailableSensors(){
        List<Sensor>sensorList = sensorManager.getSensorList(Sensor.TYPE_ALL);
        StringBuilder sensorInfo =new StringBuilder("Available sensor: \n");
        for (Sensor sensor :sensorList){
            sensorInfo.append(sensor.getName()).append("(").append(sensor.getType()).append(")\n");
        }
        sensorListTextView.setText(sensorInfo.toString());
    }


}

File: Lab2/Kotlin/ImageExperiment/app/src/main/java/com/example/imageexperiment/MainActivity.kt
package com.example.imageexperiment

import android.os.Bundle
import android.widget.Button
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var layout: RelativeLayout
    private var backgroundIndex = 0
    private val backgroundImages = listOf(
        R.drawable.background_image1,
        R.drawable.background_image2,
        R.drawable.background_image3,
        R.drawable.background_image4
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        layout = findViewById(R.id.layout)
        val switchBackgroundButton: Button = findViewById(R.id.switchBackgroundButton)

        switchBackgroundButton.setOnClickListener {
            backgroundIndex = (backgroundIndex + 1) % backgroundImages.size
            layout.setBackgroundResource(backgroundImages[backgroundIndex])
        }
    }
}
